ashley/src/main/kotlin/ktx/ashley/engine/engines.kt

```diff
@@ -0,0 +1,52 @@
+package ktx.ashley.engine
+
+import com.badlogic.ashley.core.Component
+import com.badlogic.ashley.core.Engine
+import com.badlogic.ashley.core.Entity
+import ktx.ashley.AshleyDsl
+
+/**
+ * Creates an instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @param configure inlined function with [T] as the receiver to allow additional configuration of the [Component]
+ * @return the created [Component]
+ */
+inline fun <reified T: Component> Entity.with(configure: (@AshleyDsl T).() -> Unit): T {
+  return with<T>().also(configure)
+}
+
+/**
+ * Creates a pooled instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @return the pooled [Component]
+ */
+inline fun <reified T: Component> Entity.with(): T {
+  val component = T::class.java.newInstance()
```

Comment:
You're using reflection to create a new instance of the component? This is error-prone and potentially inefficient (especially on Android), component might not have a no-arg constructor. I think you might solve it by keeping `() -> T` providers somewhere. User would have to register a provider of entity instances once - via the `Engine`? - but the same syntax for adding components could be maintained. I'm afraid this requires extending `Engine` implementations.